### ISSUE TITLE
Wallet wrongly detects BCSC app installed after going back from app store

### ISSUE THREAD

**Issue by knguyenBC** [bcgov/bc-wallet-mobile#1904]
### Issue
When the user enters the Person credential issuance screen from the BC Wallet (the screen where you start the process from BC Wallet). 
BC Wallet correctly determines that the BCSC app IS NOT installed HOWEVER when they tap to install the app and then go back to the BC Wallet without installing the BCSC app, the screen changes to wrongly determine the BCSC all was installed.

### Steps to reproduce
- Not have the BCSC app installed on the device
- Get your Person credential
- Tap "install the app", which will take the user to the app stores
- Go back to the BC Wallet app without installing the BCSC app
- User will see that Step 1 was complete, even though the BCSC app was not installed

### OS affected
Android
iOS

### Devices
iPhone 12
Google Pixel

**Comment by cvarjao** [comment]
@knguyenBC , note that this is only for non-production

**Comment by knguyenBC** [comment]
> [@knguyenBC](https://github.com/knguyenBC) , note that this is only for non-production

non-production of what? I tried with the app store version of the BC Wallet (1.0.15), without any BCSC app installed.

**Comment by jleach** [comment]
Confirmed in latest 1.0.21 build.

**Comment by jleach** [comment]
Unfortunately, due to sandboxing and security restrictions, there's no direct way in iOS to reliably check if another specific app is installed.  Apple prioritizes user privacy and app security, limiting access to information about other apps on the device.

**Workarounds (with limitations):**

- URL schemes: If the other app has a registered URL scheme, you can attempt to open it using UIApplication.shared.open(). If the app is installed, it will open; otherwise, nothing will happen or an error might occur. However, this method doesn't provide a definitive "yes/no" answer and can be unreliable.
- CanOpenURL (deprecated): This method was used in older iOS versions but is now deprecated and unreliable due to security changes.

**Comment by jleach** [comment]
The BSCS v3.10.0 has some some registered that we could use. Once the feature is enabled we can register our app to be able to query the appropriate schemas. This will allow us to check if that app is installed or not and direct people to the app store accordingly

```
<key>LSApplicationQueriesSchemes</key>
<array>
  <string>myapp</string>
  <string>anotherapp</string>
</array>
```
